### PR TITLE
Feat setup discord bot

### LINKED ISSUE #1: bot: Setup bot, list live duels
Welcome!

This is the initial kick for making the [Pistols at 10 Blocks](https://github.com/underware-gg/pistols) discord bot a reality!

This repository was cloned from [dojo.js/examples/node/torii-bot](https://github.com/dojoengine/dojo.js/tree/main/examples/node/torii-bot), which is a working implementation of a generic bot that accesses a Dojo game from its Torii server. This example was made by Loaf (@ponderingdemocritus) from the Realms/Dojo team, who is co-hosting this hack and can help if there are any issues.

## Tasks

- [ ] There's a detailed description of how to setup the bot locally on the bot [README](https://github.com/underware-gg/pistols-discord-bot/tree/main/discord-bot)
- [ ] The manifest file required by [dojoConfig.ts](https://github.com/underware-gg/pistols-discord-bot/blob/main/discord-bot/dojoConfig.ts) is here: [manifest.json](https://github.com/underware-gg/pistols/blob/main/client/src/manifest.json). Copy it to `/discord-bot`.
- [ ] Configure the bot to access Pistols at 10 Blocks Slot deployment at: `https://api.cartridge.gg/x/pistols/torii`
- [ ] Create a simple read command: `live_duels`. You must create a GraphQL query that reads the `Challenge` model, where `state = ChallengeState.InProgress`. For now, it is enough to just list the Duel Ids.

The `Challenge` [model](https://github.com/underware-gg/pistols/blob/34f83f6d34c079e8e3d52709fe56472a3176b934/dojo/src/models/models.cairo#L24-L39):
```rust
#[derive(Model, Copy, Drop, Serde)]
struct Challenge {
    #[key]
    duel_id: u128,
    //-------------------------
    duelist_a: ContractAddress, // Challenger
    duelist_b: ContractAddress, // Challenged
    message: felt252,           // message to challenged
    // progress and results
    state: u8,                  // actually a ChallengeState
    round_number: u8,           // current or final
    winner: u8,                 // 0:draw, 1:duelist_a, 2:duelist_b
    // timestamps in unix epoch
    timestamp_start: u64,       // Unix time, started
    timestamp_end: u64,         // Unix time, ended
}
```

- [ ] Create a simple settings json file that can be read by commands
- [ ] Host the bot at a Node server (we can deploy to our CloudFlare account once it's ready)
- [ ] Add the bot to a Discord server (we can provide and configure the server)

## Notes

* Playable deployment: [pistols.lootunder.world](https://pistols.lootunder.world)
* Pistols game source code (Cairo contracts + next.js client): [github.com/underware-gg/pistols](https://github.com/underware-gg/pistols)
* There is a GraphQL [playground](https://api.cartridge.gg/x/pistols/torii/graphql) to test queries
* The bot will query the game using GraphQL. Go to the [Dojo Book](https://book.dojoengine.org/toolchain/torii/graphql) for reference.
* Got questions? Talk to us! If specific about this issue, coding, Pistols, Discord, setup, etc, please comment here. For anything else, go to our [Telegram group](https://t.me/+TziqfXoXkXJiMGFk).

## Requirements

What you need to know to make this happen:

* Node
* Typescript
* GraphQL
* Be fast, as all other issues depend on this one being ready and deployed.

### PR DESCRIPTION
**Pull request by Ugo-X**
resolves #1 
## Description
{ } the discord app has been created
{ } the live_duel command has been created and put in the bot.

the discord app has been created and the bot added. we would need to move over to npm to be able to host with nodejs. Also, i would send you the secrets confidentially so you can also test out the app. ps: the env variables are not read, even with dotenv. i had to use them directly while testing,.

### CODE REVIEW

**Comment by Ugo-X** [comment]
i would send you the whole env secrets on Telegram, so please put as noted. Also

run as follows.

bun run install

then bun run codegen

then bun run build 

before you bun run serve --watch

File: discord-bot/src/config.js
Comment (rsodre) [inline discord-bot/src/config.js:1]: this file is a build product, it should not be versioned.
please delete.

File: discord-bot/src/dojoConfig.js
Comment (rsodre) [inline discord-bot/src/dojoConfig.js:1]: this file is a build product, it should not be versioned.
please delete.

File: discord-bot/src/generated/graphql.js
Comment (rsodre) [inline discord-bot/src/generated/graphql.js:1]: this file is a build product, it should not be versioned.
please delete.

File: discord-bot/src/index.js
Comment (rsodre) [inline discord-bot/src/index.js:1]: this file is a build product, it should not be versioned.
please delete.

File: discord-bot/src/queries/getMoves.js
Comment (rsodre) [inline discord-bot/src/queries/getMoves.js:1]: this file is a build product, it should not be versioned.
please delete.

File: discord-bot/src/commands/getMoves.ts
Comment (rsodre) [inline discord-bot/src/commands/getMoves.ts:1]: please rename to getChallenges.ts

File: discord-bot/src/queries/getMoves.ts
Comment (rsodre) [inline discord-bot/src/queries/getMoves.ts:1]: please rename to getChallenges.ts

File: discord-bot/src/deploy-commands.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    //     "1232006088765739049"
+    //   ),
+    //   { body: commands }
+    // );
+    const data = await rest.put(
+        Routes.applicationCommands(//application id here),
+      {
+        body: commands,
+      }
+    );
```
Comment (rsodre) [inline discord-bot/src/deploy-commands.ts]: there's a build error here

```
$ npm run build

src/deploy-commands.ts:55:6 - error TS1005: ')' expected.

55     );
        ~
Found 1 error in src/deploy-commands.ts:55
```

Missing a `)`, line 55 should be:

```
 ));
```

**Review by rsodre — APPROVED** [review]

**Pull request merged**